packages/frontend-api/src/Model/Resolver/Products/ProductsResolver.php

```diff
@@ -281,10 +288,25 @@ protected function getOrderingModeFromArgument(Argument $argument): string
     }
 
     /**
+     * @param \Overblog\GraphQLBundle\Definition\Argument $argument
      * @return string
      */
-    protected function getDefaultOrderingMode(): string
+    protected function getDefaultOrderingMode(Argument $argument): string
     {
+        if (isset($argument['search'])) {
+            return ProductListOrderingConfig::ORDER_BY_RELEVANCE;
+        }
+
         return ProductListOrderingConfig::ORDER_BY_PRIORITY;
     }
+
+    /**
+     * @param \Overblog\GraphQLBundle\Definition\Argument $argument
+     */
+    protected function checkSearchIsNotUsed(Argument $argument): void
```

Comment:
Isn't possible to create a new argument builder without a search field and use it in BrandDecorator and CategoryDecorator for products?

So it would not be necessary to make this check and the documentation would be more clear (right now I see a search field that throws error)

Reply:
c537f8d8caa654843e2fe5aade18b7793a10ec15